crates/data_components/src/token_provider.rs

```diff
@@ -6,7 +6,7 @@ use snafu::prelude::*;
 
 #[derive(Debug, Snafu)]
 pub enum Error {
-    #[snafu(display("Unable to get token: {source}"))]
+    #[snafu(display("Failed to get token.\n{source}"))]
```

Comment:
We might add an action to verify configuration and link to connector docs

Token is retrieved based on configuration

```rust
 let token_response = self
            .token_generator
            .generate_token(
                Arc::clone(&self.app_client_id),
                Arc::clone(&self.private_key),
                Arc::clone(&self.installation_id),
            )
            .await
            .map_err(|e| Error::UnableToGetToken {
                source: Box::new(e),
            })?;
```